src/dialog/an-dialog.factory.js

```diff
@@ -151,6 +160,53 @@
                         });
                     };
 
+                    var getFocusableElements = function() {
+                        var focusableEls = element.find('a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]');
```

Comment:
At the time when the dialog is opened, not necessarily all input elements are displayed yet. Also the disabled state of an input can change over time. A dialog can also contain multiple wizard steps which each contain inputs and buttons (for example the publish dialog). Which means that I think that setting focusableEls only once, when the dialog gets opened, will not work in all cases. Probably it needs to be re-evaluated every time the TAB key is used?

Reply:
Here is a possible solution to this problem:

```JavaScript
document.getElementsByTagName('body')[0]
  .addEventListener('focusin', function(event) {
      // this function gets called for each 'focusin' event which happens
      // 1. check if event.target is outside of the modal (e.g. modalElement.contain(event.target))
      // 2. if event.target is outside of the the modal then focus the first element of the modal instead
  });
```